frontend/packages/console-shared/src/components/modals/CreateProjectModal.tsx

```diff
@@ -0,0 +1,248 @@
+import * as React from 'react';
+import { Popover, Button, Modal, ModalVariant, Alert } from '@patternfly/react-core';
+import OutlinedQuestionCircleIcon from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom-v5-compat';
+import {
+  CreateProjectModal as CreateProjectModalExtension,
+  CreateProjectModalProps,
+  isCreateProjectModal,
+  useResolvedExtensions,
+} from '@console/dynamic-plugin-sdk/src';
+import { setFlag } from '@console/internal/actions/features';
+import {
+  documentationURLs,
+  ExternalLink,
+  getDocumentationURL,
+  isManaged,
+  resourceObjPath,
+  LoadingInline,
+} from '@console/internal/components/utils';
+import { ProjectRequestModel } from '@console/internal/models';
+import { k8sCreate, referenceFor } from '@console/internal/module/k8s';
+import { FLAGS } from '@console/shared';
+import { ModalComponent } from 'packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider';
+
+const DefaultCreateProjectModal: ModalComponent<CreateProjectModalProps> = ({
+  closeModal,
+  onSubmit,
+}) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [inProgress, setInProgress] = React.useState(false);
```

Comment:
Lets type these as well

Reply:
Type is inferred from the initial value for primitive values.